code/modules/mob/living/silicon/pai/pai.dm

```diff
@@ -135,6 +135,12 @@
 		icon_state = "[chassis]_dead"
 	else
 		icon_state = resting ? "[chassis]_rest" : "[chassis]"
+	update_fire()
+
+/mob/living/silicon/pai/update_fire()
+	overlays -= image("icon"='icons/mob/OnFire.dmi', "icon_state"="Generic_mob_burning")
+	if(on_fire)
+		overlays += image("icon"='icons/mob/OnFire.dmi', "icon_state"="Generic_mob_burning")
```

Comment:
this is probably best put in `update_overlays()` with an `update_icon(UPDATE_OVERLAYS)`

but ONLY if pai's don't manually add overlays elsewhere (knowing your luck they do)

Reply:
I couldnt find anywhere of pais using overlays, and it seemed to work the same. I did it in kind of a round about way because update_icons is deprecated and It feels out of scope to refactor all the update icons calls for pais. Update_fire was kept around too